internal/adapters/operation/operation_storage_redis_test.go

```diff
@@ -876,3 +876,146 @@ func TestCleanOperationsHistory(t *testing.T) {
+			client := test.GetRedisConnection(t, redisAddress)
+			clock := clockmock.NewFakeClock(time.Now())
+			operationsTTlMap := map[Definition]time.Duration{}
+			storage := NewRedisOperationStorage(client, clock, operationsTTlMap)
+
+			for _, op := range operations {
+				executionHistoryJson, err := json.Marshal(op.ExecutionHistory)
+				require.NoError(t, err)
+
+				err = client.ZAdd(context.Background(), storage.buildSchedulerHistoryOperationsKey(op.SchedulerName), &redis.Z{
+					Member: op.ID,
+					Score:  float64(op.CreatedAt.Unix()),
+				}).Err()
+				require.NoError(t, err)
+
+				for _, op := range operations[0:1] {
+					err = client.HSet(context.Background(), storage.buildSchedulerOperationKey(op.SchedulerName, op.ID), map[string]interface{}{
+						idRedisKey:                 op.ID,
+						schedulerNameRedisKey:      op.SchedulerName,
+						statusRedisKey:             strconv.Itoa(int(op.Status)),
+						definitionNameRedisKey:     op.DefinitionName,
+						createdAtRedisKey:          op.CreatedAt.Format(time.RFC3339Nano),
+						definitionContentsRedisKey: op.Input,
+						executionHistoryRedisKey:   executionHistoryJson,
+					}).Err()
+					require.NoError(t, err)
+				}
+
+				err = storage.CleanExpiredOperations(context.Background(), schedulerName)
+				assert.NoError(t, err)
+
+				operationIds, err := client.ZRangeByScore(context.Background(), fmt.Sprintf("operations:%s:lists:history", schedulerName), &redis.ZRangeBy{
+					Min: "-inf",
+					Max: "+inf",
+				}).Result()
+				require.NoError(t, err)
+				assert.NotEmpty(t, operationIds)
+				assert.Contains(t, operationIds, operations[0].ID, operations[1].ID)
+				assert.NotContains(t, operationIds, operations[2].ID, operations[3].ID)
+			}
+		})
+
+		t.Run("if operations history is empty it returns with no error", func(t *testing.T) {
+			client := test.GetRedisConnection(t, redisAddress)
+			clock := clockmock.NewFakeClock(time.Now())
+			operationsTTlMap := map[Definition]time.Duration{}
+			storage := NewRedisOperationStorage(client, clock, operationsTTlMap)
+
+			operationsReturned, err := storage.ListSchedulerFinishedOperations(context.Background(), schedulerName)
+			assert.NoError(t, err)
+			assert.Empty(t, operationsReturned)
+
+			err = storage.CleanExpiredOperations(context.Background(), schedulerName)
+
+			assert.NoError(t, err)
+		})
+
+		t.Run("with error", func(t *testing.T) {
+
+			t.Run("if client is closed it returns error", func(t *testing.T) {
+				client := test.GetRedisConnection(t, redisAddress)
+				clock := clockmock.NewFakeClock(time.Now())
+				operationsTTlMap := map[Definition]time.Duration{}
+				storage := NewRedisOperationStorage(client, clock, operationsTTlMap)
+				client.Close()
+
+				err := storage.CleanExpiredOperations(context.Background(), schedulerName)
+
+				assert.EqualError(t, err, "failed to list operations for \"test-scheduler\": redis: client is closed")
```

Comment:
```suggestion
assert.ErrorContains(t, err, "failed to list operations for \"test-scheduler\":")
```

Another comment here is that `failed to list operations for` when I see this error on the logs I will not correlate it to a cleanup function, it is more related to list operations for example `GetOperation`, `ListSCHedulerActiveOperations`, etc. What do you think about improving this log message by adding something that connects it with the `CleanExpiredOperations` function?